otcextensions/sdk/dds/v3/instance.py

```diff
@@ -0,0 +1,207 @@
+    start_time = resource.Body('start_time')
+    #: Specifies the number of days to retain the generated backup files.
+    #: valid value range is from 0 to 732.
+    #: *Type:string*
+    keep_days = resource.Body('keep_days')
+
+
+class DatastoreSpec(resource.Resource):
+    #: Specifies the database type.
+    #: The value is ``DDS-Community``.
+    #: *Type:string*
+    type = resource.Body('type')
+    #: Specifies the database version.
+    #: valid value is 3.2 or 3.4.
+    #: *Type:string*
+    version = resource.Body('version')
+    #: Specifies the storage engine.
+    #: valid value is wiredTiger.
+    #: *Type:string*
+    storage_engine = resource.Body('storage_engine')
+
+
+class Instance(resource.Resource):
+    base_path = '/instances'
+
+    resources_key = 'instances'
+    resource_key = 'instance'
+
+    # capabilities
+    allow_create = True
+    allow_delete = True
+    allow_commit = True
+    allow_list = True
+    allow_fetch = True
+
+    _query_mapping = resource.QueryParameters(
+        'id', 'name', 'mode', 'datastore_type',
+        'vpc_id', 'subnet_id', 'limit', 'offset')
+
+    #: DB instance ID.
+    #: *Type:string*
+    id = resource.Body('id')
+    #: DB instance name.
+    #: *Type:string*
+    name = resource.Body('name')
+    #: Specifies the instance type.
+    #: * Sharding indicates the cluster instance.
+    #: * ReplicaSet indicate the replica set instance.
+    #: *Type:string*
+    mode = resource.Body('mode')
+    #: Specifies the database type.
+    #: *Type:string*
+    datastore_type = resource.Body('datastore_type')
+    #: Specifies the VPC ID.
+    #: *Type:string*
+    vpc_id = resource.Body('vpc_id')
+    #: Data store information.
+    #: *Type:string*
+    subnet_id = resource.Body('subnet_id')
+    #: Specifies the ID of the security group
+    #: where a specified DB instance belongs to.
+    #: *Type:string*
+    security_group_id = resource.Body('security_group_id')
+    #: Specifies the database information.
+    #: *Type:dict*
+    datastore = resource.Body('datastore', type=DatastoreSpec)
+    #: Specifies the region ID.
+    #: *Type:string*
+    region = resource.Body('region')
+    #: Specifies the AZ ID.
+    #: *Type:string*
+    availability_zone = resource.Body('availability_zone')
+    #: Specifies the database password.
+    #: *Type:string*
+    password = resource.Body('password')
+    #: Specifies the key ID used for disk encryption.
+    #: *Type:string*
+    disk_encryption_id = resource.Body('disk_encryption_id')
+    #: Specifies the instance specifications.
+    #: *Type:list*
+    flavor = resource.Body('flavor', type=list, list_type=FlavorSpec)
+    #: Specifies the advanced backup policy.
+    #: *Type:dict*
+    backup_strategy = resource.Body('backup_strategy', type=BackupStrategySpec)
+    #: Specifies whether to enable SSL.
+    #: *Type:string*
+    ssl_option = resource.Body('ssl_option')
+
+    def fetch(self, session, requires_id=True,
```

Comment:
why do you overload this function? If to "simulate" GET then why so complex? https://github.com/opentelekomcloud/python-otcextensions/blob/master/otcextensions/sdk/rds/v3/instance.py#L229 does same a bit shorter. If there is a reason maybe you add some comments?

Reply:
it caused by wrong get_instance func from proxy, now fixed

Reply:
still don't get it.
- service does not implement GET method, therefore we want to simulate it by doing find inside
- list + get_one_match can be implemented as in referred RDS instance case
- you repeat query parameters here (same as defined in the class itself)
- most of query params would be send anyway empty, since regular case is to implement conn.dds.get_instance('my_id')
- you reimplement response parsing

Overall I see to much repeated logic without understanding necessity for it (assuming it is really necessary)

Reply:
sry, didn't pushed change